python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1964 @@
+                metadata['boresight_dec_rad'] = float('nan')
+        except Exception as e:
+            self.log.debug("Could not extract boresightRaDec: %s", e)
+            metadata['boresight_ra_rad'] = float('nan')
+            metadata['boresight_dec_rad'] = float('nan')
+
+        # MJD date
+        try:
+            exp_date = getattr(visit_info, 'date', None)
+            if exp_date is not None:
+                # Convert to astropy and get MJD
+                mjd = exp_date.toAstropy().tai.mjd
+                metadata['mjd'] = mjd
+            else:
+                metadata['mjd'] = float('nan')
+        except Exception as e:
+            self.log.debug("Could not extract date: %s", e)
+            metadata['mjd'] = float('nan')
+
+        return metadata
+
+    def createDonutStampFromTarts(
+        self, exposure: afwImage.Exposure, cropped_image: np.ndarray, defocalType: str
+    ) -> DonutStamps:
+        """Create DonutStamps from TARTS output - handles multiple donuts."""
+        # Log image type and shape for debugging
+        try:
+            image_shape = cropped_image.shape
+            self.log.debug(
+                f"Creating DonutStamps; input image shape: {image_shape}, defocalType='{defocalType}'"
+            )
+        except AttributeError:
+            self.log.debug(
+                f"Creating DonutStamps; input image type: {type(cropped_image)}, defocalType='{defocalType}'"
+            )
+        # Extract information from exposure
+        detector = exposure.getDetector()
+        detectorName = detector.getName()
+        cameraName = "LSSTCam"
+        bandLabel = exposure.filter.bandLabel
+
+        # Get exposure dimensions for centroid calculation
+        bbox = exposure.getBBox()
+        center_x = bbox.getCenterX()
+        center_y = bbox.getCenterY()
+
+        # Use the cropped image directly as it's guaranteed to be a numpy array
+        image_array = cropped_image
+
+        # Handle different input shapes
+        if len(image_array.shape) == 2:
+            # Single donut: [cropSize, cropSize] -> [1, cropSize, cropSize]
+            image_array = image_array.reshape(1, self.cropSize, self.cropSize)
+        elif len(image_array.shape) == 3:
+            # Multiple donuts: [num_stamps, cropSize, cropSize] - correct shape
+            pass
+        else:
+            raise ValueError(f"Unexpected image array shape: {image_array.shape}")
+
+        num_stamps = image_array.shape[0]
+        self.log.info("Processing %d donut stamp(s) of size %dx%d pixels",
+                     num_stamps, self.cropSize, self.cropSize)
+        self.log.debug("DONUT PROCESSING DEBUG: Starting donut stamp creation")
+        self.log.debug("Normalized to %d stamp(s) of size %dx%d", num_stamps, self.cropSize, self.cropSize)
+        # Get centroid positions using helper method
+        cent_x_list, cent_y_list = self._calculate_centroid_positions(exposure, num_stamps)
```

Comment:
So you are getting the centroids using exposure and number of stamps. Is that regardless of what the cropped image is? I would have thought the cropped image would have that information, instead of having to handle this here.